test/object-store/c_api/c_api.cpp

```diff
@@ -449,6 +449,99 @@ TEST_CASE("C API (non-database)", "[c_api]") {
+            realm_app_config_set_base_url(app_config.get(), "https://path/to/app");
+            CHECK(app_config->base_url == "https://path/to/app");
+        }
+
+        SECTION("realm_app_config_set_local_app_name()") {
+            realm_app_config_set_local_app_name(app_config.get(), "some_app_name");
+            CHECK(app_config->local_app_name == "some_app_name");
+        }
+
+        SECTION("realm_app_config_set_local_app_version()") {
+            realm_app_config_set_local_app_version(app_config.get(), "some_app_version");
+            CHECK(app_config->local_app_version == "some_app_version");
+        }
+
+        SECTION("realm_app_config_set_default_request_timeout()") {
+            realm_app_config_set_default_request_timeout(app_config.get(), 2500);
+            CHECK(app_config->default_request_timeout_ms == 2500);
+        }
+
+        SECTION("realm_app_config_set_platform()") {
+            realm_app_config_set_platform(app_config.get(), "some_platform_name");
+            CHECK(app_config->device.platform == "some_platform_name");
+        }
+
+        SECTION("realm_app_config_set_platform_version()") {
+            realm_app_config_set_platform_version(app_config.get(), "some_platform_version");
+            CHECK(app_config->device.platform_version == "some_platform_version");
+        }
+
+        SECTION("realm_app_config_set_sdk_version()") {
+            realm_app_config_set_sdk_version(app_config.get(), "some_sdk_version");
+            CHECK(app_config->device.sdk_version == "some_sdk_version");
+        }
+
+        SECTION("realm_app_config_set_sdk()") {
+            realm_app_config_set_sdk(app_config.get(), "some_sdk_name");
+            CHECK(app_config->device.sdk == "some_sdk_name");
+        }
+
+        SECTION("realm_app_config_set_cpu_arch()") {
+            realm_app_config_set_cpu_arch(app_config.get(), "some_cpu_arch");
+            CHECK(app_config->device.cpu_arch == "some_cpu_arch");
+        }
+
+        SECTION("realm_app_config_set_device_name()") {
+            realm_app_config_set_device_name(app_config.get(), "some_device_name");
+            CHECK(app_config->device.device_name == "some_device_name");
+        }
+
+        SECTION("realm_app_config_set_device_version()") {
+            realm_app_config_set_device_version(app_config.get(), "some_device_version");
+            CHECK(app_config->device.device_version == "some_device_version");
+        }
+
+        SECTION("realm_app_config_set_framework_name()") {
+            realm_app_config_set_framework_name(app_config.get(), "some_framework_name");
+            CHECK(app_config->device.framework_name == "some_framework_name");
+        }
+
+        SECTION("realm_app_config_set_framework_version()") {
+            realm_app_config_set_framework_version(app_config.get(), "some_framework_version");
+            CHECK(app_config->device.framework_version == "some_framework_version");
+        }
+    }
```

Comment:
Do we want to try actually connecting to a baas app and making sure the values we pass in here actually propagate back to the server correctly?

Reply:
I could create a dummy test transport like `UnitTestTransport` in test/object-store/app.cpp and verify the device json sent with login.

Reply:
What I meant was more, should we actually do an E2E test that makes sure that these values actually show up in the database on the server side. If the answer is no then can we do some manual QA to make sure they actually show up in the database on the server side. This test case is very unit testy, I'm asking if we should build an integration test.

Reply:
I don't think we need to worry about verifying the server side - the new values aren't currently being handled on the server side and are ignored. I just spoke with Kiro and some values are stored in the DB for auth (accessible via `/api/admin/v3.0/groups/%s/apps/%s/users/%s/devices`) and it is likely most of the new values are only going to be pushed to Splunk, since the devices table isn't really meant for metrics/history information.